Modules/Core/Common/src/itkMultiThreaderBase.cxx

```diff
@@ -217,8 +217,7 @@ MultiThreaderBase::SetGlobalMaximumNumberOfThreads(ThreadIdType val)
   m_PimplGlobals->m_GlobalMaximumNumberOfThreads = std::clamp<ThreadIdType>(val, 1, ITK_MAX_THREADS);
 
   // If necessary reset the default to be used from now on.
-  m_PimplGlobals->m_GlobalDefaultNumberOfThreads =
-    std::min(m_PimplGlobals->m_GlobalDefaultNumberOfThreads, m_PimplGlobals->m_GlobalMaximumNumberOfThreads);
+  MultiThreaderBase::SetGlobalDefaultNumberOfThreads(std::min(GetGlobalMaximumNumberOfThreads(), m_PimplGlobals->m_GlobalMaximumNumberOfThreads));
```

Comment:
Thanks Matt, but is this as you intended? 

    std::min(GetGlobalMaximumNumberOfThreads(), m_PimplGlobals->m_GlobalMaximumNumberOfThreads))

That's just `min(Maximum, Maximum)`, right?

Reply:
So it unconditionally does `Default = Maximum`?

Reply:
That is a good point. Perhaps we should reuse the setter method too:
```
Self::SetGlobalDefaultNumberOfThreads(Self::GetGlobalDefaultNumberOfThreads())
```

Reply:
Just for my understanding, what is the difference between the `GlobalDefault` and the `GlobalMaximum`, conceptually? When would a user set the `GlobalDefault`, and when would a user set the `GlobalMaximum`?

It seems to me that the `GlobalDefault` is the most important one, as in is used as the number of threads of the thread pool.

Reply:
Oops, yes it should be min(Default, Maximum).

When would a user set the GlobalDefault, and when would a user set the GlobalMaximum?

The Default defines the default value and the maximum defines a limit.

Reply:
> The Default defines the default value and the maximum defines a limit.

OK, thanks, but isn't the number of threads of the thread pool all that matters?

Reply:
@N-Dekker what do you mean?

Reply:
I mean, does ITK really need two global variables to specify the number of threads? Wouldn't it be sufficient to just have one GlobalNumberOfThreads variable?

Reply:
I think the intention is to support a limit on the number of threads that can be requested that is separate from the number of threads used by a filter.

Reply:
@N-Dekker an example use case may help:

- The maximum number of threads is 128, the number of cores on a many-core system.
- This is the maximum number of threads, but the default number of threads may be set to 12, which will result in better utilization of resources when processing a 256x256 image or when a simple filter is applied.
- Yet, more threads could be used where there is larger data or a more computationally complex filter.